src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -280,10 +282,13 @@ private ParsedDocument parseRequest(IndexService documentIndexService, Percolate
                         }
 
                         MapperService mapperService = documentIndexService.mapperService();
-                        Tuple<DocumentMapper, Boolean> docMapper = mapperService.documentMapperWithAutoCreate(request.documentType());
-                        doc = docMapper.v1().parse(source(parser).type(request.documentType()).flyweight(true)).setMappingsModified(docMapper);
-                        if (doc.mappingsModified()) {
-                            mappingUpdatedAction.updateMappingOnMaster(request.shardId().getIndex(), docMapper.v1(), documentIndexService.indexUUID());
+                        Tuple<DocumentMapper, Mapping> docMapper = mapperService.documentMapperWithAutoCreate(request.documentType());
+                        doc = docMapper.v1().parse(source(parser).type(request.documentType()).flyweight(true));
+                        if (docMapper.v2() != null) {
+                            doc.addDynamicMappingsUpdate(docMapper.v2());
+                        }
+                        if (doc.dynamicMappingsUpdate() != null) {
```

Comment:
I don't think we can? It might happen that the first condition is false and the second is true if you only got dynamic updates through `copy_to` directives (which needs to be handled differently since it can insert dynamic mappings at arbitrary places in the mappings).